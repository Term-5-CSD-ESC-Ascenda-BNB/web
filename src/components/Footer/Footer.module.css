.footer {
  width: 100%;
  min-height: 250px;
  background-color: #43455e;
  display: flex;
  align-items: flex-start;
  justify-content: space-between;
  padding: 40px 5% 40px 5%;
  gap: 40px;
}

.logo-container {
  flex: 1;
  max-width: 280px;
  align-items: flex-start;
  justify-content: flex-start;
  padding-top: 10px;
}

.footer-logo {
  color: white;
}

.copyright {
  color: #b8bcc8;
  font-size: 14px;
  margin: 0;
  line-height: 1.4;
}

.details-container {
  flex: 1;
  max-width: 200px;
  align-items: flex-start;
  justify-content: flex-start;
  padding-top: 10px;
}

.section-title {
  color: white;
  font-size: 16px;
  font-weight: 600;
  margin: 0 0 3px 0;
  letter-spacing: 0.5px;
}

.line {
  width: 80px;
  height: 1px;
  background-color: white;
  margin: 0;
}

.contact-info p,
.nav-links p,
.legal-links p {
  color: #b8bcc8;
  font-size: 14px;
  margin: 8px 0;
  line-height: 1.4;
  cursor: pointer;
  transition: color 0.2s ease;
}

.contact-info p:hover,
.nav-links p:hover,
.legal-links p:hover {
  color: white;
}

.social-icons {
  display: flex;
  gap: 12px;
  margin-top: 16px;
}

.social-icon {
  width: 36px;
  height: 36px;
  background-color: rgba(255, 255, 255, 0.1);
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  color: white;
  font-size: 12px;
  font-weight: bold;
  cursor: pointer;
  transition: all 0.2s ease;
}

.social-icon:hover {
  background-color: rgba(255, 255, 255, 0.2);
  transform: translateY(-2px);
}

.contact-info p {
  display: flex;
  align-items: center;
  gap: 8px;
}
