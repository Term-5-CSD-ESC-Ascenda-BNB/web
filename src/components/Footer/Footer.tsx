import { Logo } from '../Logo/Logo';
import styles from './Footer.module.css';
import { Stack } from '@mantine/core';
import {
  IconBrandFacebook,
  IconBrandTiktok,
  IconBrandX,
  IconBrandInstagram,
  IconPhone,
  IconMail,
} from '@tabler/icons-react';

export function Footer() {
  return (
    <div className={styles['footer']}>
      <Stack gap={20} className={styles['logo-container']}>
        <Logo c={'gray.1'} />
        <p className={styles['copyright']}>Â© 2025 Wayfare. All rights reserved.</p>
      </Stack>

      <Stack gap={16} className={styles['details-container']}>
        <h3 className={styles['section-title']}>Contact Us</h3>
        <div className={styles['line']}></div>
        <div className={styles['contact-info']}>
          <p>
            <IconMail size={18} /> contact@wayfare.com
          </p>
          <p>
            <IconPhone size={18} /> +65 6789 1234
          </p>
        </div>
        <div className={styles['social-icons']}>
          <div className={styles['social-icon']}>
            <IconBrandFacebook size={18} />
          </div>
          <div className={styles['social-icon']}>
            <IconBrandTiktok size={18} />
          </div>
          <div className={styles['social-icon']}>
            <IconBrandX size={18} />
          </div>
          <div className={styles['social-icon']}>
            <IconBrandInstagram size={18} />
          </div>
        </div>
      </Stack>

      <Stack gap={16} className={styles['details-container']}>
        <h3 className={styles['section-title']}>Navigation</h3>
        <div className={styles['line']}></div>
        <div className={styles['nav-links']}>
          <p>About Us</p>
          <p>Hotels</p>
          <p>FAQs</p>
          <p>My Bookings</p>
        </div>
      </Stack>

      <Stack gap={16} className={styles['details-container']}>
        <h3 className={styles['section-title']}>Legal</h3>
        <div className={styles['line']}></div>
        <div className={styles['legal-links']}>
          <p>Terms of Service</p>
          <p>Privacy Policy</p>
          <p>Cookie Policy</p>
        </div>
      </Stack>
    </div>
  );
}
