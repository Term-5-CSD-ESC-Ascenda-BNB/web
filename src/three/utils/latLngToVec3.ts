import * as THREE from 'three';

export function latLngToVec3(lat: number, lng: number, radius: number): THREE.Vector3 {
  const elevation = THREE.MathUtils.degToRad(90 - lat);
  const azimuth = THREE.MathUtils.degToRad(lng) + Math.PI / 2; // Add PI/2 to align with the globe's coordinate system

  const position = new THREE.Vector3().setFromSphericalCoords(radius, elevation, azimuth);

  return position;
}
