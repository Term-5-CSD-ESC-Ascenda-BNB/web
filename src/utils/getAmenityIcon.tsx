import {
  IconWind,
  IconBriefcase,
  IconIroning3,
  IconPlugConnected,
  IconShirt,
  IconScissors,
  IconBath,
  IconUsersGroup,
  IconSwimming,
  IconParking,
  IconBell,
  IconScale,
  IconDeviceTv,
  IconPhone,
  IconWriting,
  IconCoffee,
  IconBed,
  IconBabyCarriage,
  IconShoe,
  IconHelpCircle,
  IconBread,
  IconToolsKitchen2,
  IconHanger2,
} from '@tabler/icons-react';
import React from 'react';

export function getAmenityIcon(label: string): React.ReactElement {
  // Normalize: lowercase, trim, collapse whitespace, fix common format issues
  const f = label
    .toLowerCase()
    .trim()
    .replace(/\s+/g, ' ')
    .replace(/t\s*v/, 'tv') // normalize "T V" to "tv"
    .replace(/voice ?mail/, 'voicemail'); // normalize variations of "voice mail"

  if (f.includes('hair')) return <IconScissors size={16} />;
  if (f.includes('air')) return <IconWind size={16} />;
  if (f.includes('business')) return <IconBriefcase size={16} />;
  if (f.includes('iron')) return <IconIroning3 size={16} />;
  if (f.includes('data') || f.includes('port')) return <IconPlugConnected size={16} />;
  if (f.includes('dry')) return <IconShirt size={16} />;
  if (f.includes('meeting')) return <IconUsersGroup size={16} />;
  if (f.includes('pool')) return <IconSwimming size={16} />;
  if (f.includes('garage') || f.includes('parking')) return <IconParking size={16} />;
  if (f.includes('room service')) return <IconBell size={16} />;
  if (f.includes('safe')) return <IconScale size={16} />;
  if (f.includes('tv')) return <IconDeviceTv size={16} />;
  if (f.includes('phone') || f.includes('voicemail')) return <IconPhone size={16} />;
  if (f.includes('desk') || f.includes('work') || f.includes('workspace'))
    return <IconWriting size={16} />;
  if (f.includes('coffee') || f.includes('tea') || f.includes('minibar') || f.includes('kettle'))
    return <IconCoffee size={16} />;
  if (f.includes('breakfast')) return <IconBread size={16} />;
  if (f.includes('kitchen')) return <IconToolsKitchen2 size={16} />;
  if (f.includes('crib') || f.includes('infant')) return <IconBabyCarriage size={16} />;
  if (f.includes('bed') || f.includes('bedding') || f.includes('linen'))
    return <IconBed size={16} />;
  if (f.includes('towel')) return <IconHanger2 size={16} />;
  if (f.includes('bath') || f.includes('soak')) return <IconBath size={16} />;
  if (f.includes('shower')) return <IconShoe size={16} />;

  return <IconHelpCircle size={16} />;
}
