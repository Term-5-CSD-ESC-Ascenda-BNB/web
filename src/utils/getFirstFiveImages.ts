import type { ImageDetails } from '@/types/Hotel';

/**
 * Extracts the first five image URLs from a hotel object.
 * @param imageDetails - The image details object containing prefix, suffix, and count.
 * @return An array of up to five image URLs.
 */
export function getFirstFiveImages(imageDetails: ImageDetails): string[] {
  if (!imageDetails || !imageDetails.prefix || !imageDetails.suffix) {
    return [];
  }

  const images: string[] = [];
  const count = Math.min(imageDetails.count, 5);

  for (let i = 0; i < count; i++) {
    images.push(`${imageDetails.prefix}${i}${imageDetails.suffix}`);
  }

  return images;
}
