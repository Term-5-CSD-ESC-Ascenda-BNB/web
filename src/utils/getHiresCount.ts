export function getHiresCount(hires_string: string): number {
  // separate the string by commas
  const hiresArray = hires_string
    .split(',')
    // filter out any empty strings
    .filter((value) => value.trim() !== '');
  // return the length of the array
  return hiresArray.length;
}
