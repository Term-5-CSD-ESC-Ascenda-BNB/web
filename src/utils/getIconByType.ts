import React from 'react';
import {
  IconBuildingSkyscraper,
  IconPlane,
  IconBuilding,
  IconBuildingBank,
  IconSchool,
  IconToolsKitchen2,
  IconShoppingBag,
  IconBuildingMonument,
  IconMask,
  IconBuildingStadium,
  IconCards,
  IconGolf,
  IconGlassFullFilled,
  IconBriefcase,
  IconBuildingHospital,
  IconBuildingCommunity,
  IconSkiJumping,
  IconAnchor,
  IconDirectionsFilled,
  IconTree,
  IconIceCream,
  IconSunglasses,
  IconBuildingEstate,
  IconMapPin,
  type IconProps,
} from '@tabler/icons-react';

/**
 * Returns the appropriate Tabler icon component for a destination type
 * @param type - The destination type (e.g., 'city', 'airport', 'museum')
 * @returns React component for the icon
 */
export const getIconByType = (type: string): React.FC<IconProps> => {
  const iconMap: Record<string, React.FC<IconProps>> = {
    city: IconBuildingSkyscraper,
    airport: IconPlane,
    hotel: IconBuilding,
    museum: IconBuildingBank,
    museums: IconBuildingBank,
    school: IconSchool,
    restaurant: IconToolsKitchen2,
    shopping: IconShoppingBag,
    historic: IconBuildingMonument,
    monument: IconBuildingMonument,
    theater: IconMask,
    stadium: IconBuildingStadium,
    casino: IconCards,
    golf: IconGolf,
    winery: IconGlassFullFilled,
    business: IconBriefcase,
    medical: IconBuildingHospital,
    civic: IconBuildingCommunity,
    skiing: IconSkiJumping,
    anchor: IconAnchor,
    sign: IconDirectionsFilled,
    tree: IconTree,
    icecream: IconIceCream,
    sunglass: IconSunglasses,
    district: IconBuildingEstate,
  };
  return iconMap[type.toLowerCase()] || IconMapPin;
};
