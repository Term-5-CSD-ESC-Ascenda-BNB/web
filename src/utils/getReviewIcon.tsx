import {
  IconMapPin,
  IconBed,
  IconWifi,
  IconCoffee,
  IconToolsKitchen2,
  IconArmchair,
  IconHeartHandshake,
  IconMoodSmile,
  IconHelpCircle,
  IconUser,
  IconUsersGroup,
  IconBriefcase,
  IconHeart,
} from '@tabler/icons-react';
import React from 'react';

export function getReviewIcon(label: string | undefined): React.ReactElement {
  if (!label) return <IconHelpCircle size={16} />;

  const l = label.toLowerCase();

  // Category scores
  if (l.includes('location')) return <IconMapPin size={16} />;
  if (l.includes('room')) return <IconBed size={16} />;
  if (l.includes('wifi') || l.includes('internet')) return <IconWifi size={16} />;
  if (l.includes('breakfast')) return <IconCoffee size={16} />;
  if (l.includes('food') || l.includes('dining')) return <IconToolsKitchen2 size={16} />;
  if (l.includes('comfort')) return <IconArmchair size={16} />;
  if (l.includes('service')) return <IconHeartHandshake size={16} />;
  if (l.includes('vibe')) return <IconMoodSmile size={16} />;

  // Traveler types
  if (l.includes('solo')) return <IconUser size={16} />;
  if (l.includes('couple')) return <IconHeart size={16} />;
  if (l.includes('family')) return <IconUsersGroup size={16} />;
  if (l.includes('business')) return <IconBriefcase size={16} />;

  return <IconHelpCircle size={16} />;
}
