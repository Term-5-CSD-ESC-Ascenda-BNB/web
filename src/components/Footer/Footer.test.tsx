import { render, screen } from '@/tests/utils';
import { describe, it, expect, vi } from 'vitest';
import { Footer } from './Footer';

// Mock @tanstack/react-router
const mockNavigate = vi.fn();
vi.mock('@tanstack/react-router', () => ({
  useRouter: () => ({ navigate: mockNavigate }),
}));

describe('Footer (presentational)', () => {
  it('renders without crashing and matches snapshot', () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });

  it('displays copyright notice', () => {
    render(<Footer />);
    expect(screen.getByText('Â© 2025 Wayfare. All rights reserved.')).toBeInTheDocument();
  });

  it('shows contact section with email and phone', () => {
    render(<Footer />);
    expect(screen.getByText('Contact Us')).toBeInTheDocument();
    expect(screen.getByText(/contact@wayfare\.com/)).toBeInTheDocument();
    expect(screen.getByText(/\+65 6789 1234/)).toBeInTheDocument();
  });

  it('lists navigation links', () => {
    render(<Footer />);
    ['About Us', 'Hotels', 'FAQs', 'My Bookings'].forEach((text) => {
      expect(screen.getByText(text)).toBeInTheDocument();
    });
  });

  it('lists legal links', () => {
    render(<Footer />);
    ['Terms of Service', 'Privacy Policy', 'Cookie Policy'].forEach((text) => {
      expect(screen.getByText(text)).toBeInTheDocument();
    });
  });
});
