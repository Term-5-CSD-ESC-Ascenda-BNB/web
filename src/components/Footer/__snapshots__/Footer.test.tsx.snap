// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer (presentational) > renders without crashing and matches snapshot 1`] = `
<div>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-font-family: DM Sans, sans-serif;--mantine-font-family-headings: Libre Franklin, serif;--mantine-primary-color-filled: var(--mantine-color-primary-filled);--mantine-primary-color-filled-hover: var(--mantine-color-primary-filled-hover);--mantine-primary-color-light: var(--mantine-color-primary-light);--mantine-primary-color-light-hover: var(--mantine-color-primary-light-hover);--mantine-primary-color-light-color: var(--mantine-color-primary-light-color);--mantine-primary-color-0: var(--mantine-color-primary-0);--mantine-primary-color-1: var(--mantine-color-primary-1);--mantine-primary-color-2: var(--mantine-color-primary-2);--mantine-primary-color-3: var(--mantine-color-primary-3);--mantine-primary-color-4: var(--mantine-color-primary-4);--mantine-primary-color-5: var(--mantine-color-primary-5);--mantine-primary-color-6: var(--mantine-color-primary-6);--mantine-primary-color-7: var(--mantine-color-primary-7);--mantine-primary-color-8: var(--mantine-color-primary-8);--mantine-primary-color-9: var(--mantine-color-primary-9);--mantine-color-primary-0: #F5F6F9;--mantine-color-primary-1: #E8EAF1;--mantine-color-primary-2: #BAC2D6;--mantine-color-primary-3: #99A4C1;--mantine-color-primary-4: #808AB1;--mantine-color-primary-5: #6E74A2;--mantine-color-primary-6: #626693;--mantine-color-primary-7: #545579;--mantine-color-primary-8: #43455E;--mantine-color-primary-9: #2E2E3D;--mantine-color-secondary-0: #F8F6F4;--mantine-color-secondary-1: #DCD6CC;--mantine-color-secondary-2: #C8BEB0;--mantine-color-secondary-3: #AD9D8A;--mantine-color-secondary-4: #9C8773;--mantine-color-secondary-5: #8F7867;--mantine-color-secondary-6: #786256;--mantine-color-secondary-7: #635249;--mantine-color-secondary-8: #51433D;--mantine-color-secondary-9: #2B231F;}:root[data-mantine-color-scheme="dark"]{--mantine-color-anchor: var(--mantine-color-primary-4);--mantine-color-primary-text: var(--mantine-color-primary-4);--mantine-color-primary-filled: var(--mantine-color-primary-8);--mantine-color-primary-filled-hover: var(--mantine-color-primary-9);--mantine-color-primary-light: rgba(98, 102, 147, 0.15);--mantine-color-primary-light-hover: rgba(98, 102, 147, 0.2);--mantine-color-primary-light-color: var(--mantine-color-primary-3);--mantine-color-primary-outline: var(--mantine-color-primary-4);--mantine-color-primary-outline-hover: rgba(128, 138, 177, 0.05);--mantine-color-secondary-text: var(--mantine-color-secondary-4);--mantine-color-secondary-filled: var(--mantine-color-secondary-8);--mantine-color-secondary-filled-hover: var(--mantine-color-secondary-9);--mantine-color-secondary-light: rgba(120, 98, 86, 0.15);--mantine-color-secondary-light-hover: rgba(120, 98, 86, 0.2);--mantine-color-secondary-light-color: var(--mantine-color-secondary-3);--mantine-color-secondary-outline: var(--mantine-color-secondary-4);--mantine-color-secondary-outline-hover: rgba(156, 135, 115, 0.05);}:root[data-mantine-color-scheme="light"]{--mantine-color-anchor: var(--mantine-color-primary-6);--mantine-color-primary-text: var(--mantine-color-primary-filled);--mantine-color-primary-filled: var(--mantine-color-primary-6);--mantine-color-primary-filled-hover: var(--mantine-color-primary-7);--mantine-color-primary-light: rgba(98, 102, 147, 0.1);--mantine-color-primary-light-hover: rgba(98, 102, 147, 0.12);--mantine-color-primary-light-color: var(--mantine-color-primary-6);--mantine-color-primary-outline: var(--mantine-color-primary-6);--mantine-color-primary-outline-hover: rgba(98, 102, 147, 0.05);--mantine-color-secondary-text: var(--mantine-color-secondary-filled);--mantine-color-secondary-filled: var(--mantine-color-secondary-6);--mantine-color-secondary-filled-hover: var(--mantine-color-secondary-7);--mantine-color-secondary-light: rgba(120, 98, 86, 0.1);--mantine-color-secondary-light-hover: rgba(120, 98, 86, 0.12);--mantine-color-secondary-light-color: var(--mantine-color-secondary-6);--mantine-color-secondary-outline: var(--mantine-color-secondary-6);--mantine-color-secondary-outline-hover: rgba(120, 98, 86, 0.05);}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="_footer_acfd2c"
  >
    <div
      class="_logo-container_acfd2c m_6d731127 mantine-Stack-root"
      style="--stack-gap: calc(1.25rem * var(--mantine-scale)); --stack-align: stretch; --stack-justify: flex-start;"
    >
      <a
        class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
        role="link"
        style="cursor: pointer; color: var(--mantine-color-gray-1); font-family: "Cormorant Garamond", serif; font-size: 2rem; font-weight: 500;"
      >
        Wayfare
      </a>
      <p
        class="_copyright_acfd2c"
      >
        Â© 2025 Wayfare. All rights reserved.
      </p>
    </div>
    <div
      class="_details-container_acfd2c m_6d731127 mantine-Stack-root"
      style="--stack-gap: calc(1rem * var(--mantine-scale)); --stack-align: stretch; --stack-justify: flex-start;"
    >
      <h3
        class="_section-title_acfd2c"
      >
        Contact Us
      </h3>
      <div
        class="_line_acfd2c"
      />
      <div
        class="_contact-info_acfd2c"
      >
        <p>
          <svg
            class="tabler-icon tabler-icon-mail "
            fill="none"
            height="18"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 7a2 2 0 0 1 2 -2h14a2 2 0 0 1 2 2v10a2 2 0 0 1 -2 2h-14a2 2 0 0 1 -2 -2v-10z"
            />
            <path
              d="M3 7l9 6l9 -6"
            />
          </svg>
           contact@wayfare.com
        </p>
        <p>
          <svg
            class="tabler-icon tabler-icon-phone "
            fill="none"
            height="18"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 4h4l2 5l-2.5 1.5a11 11 0 0 0 5 5l1.5 -2.5l5 2v4a2 2 0 0 1 -2 2a16 16 0 0 1 -15 -15a2 2 0 0 1 2 -2"
            />
          </svg>
           +65 6789 1234
        </p>
      </div>
      <div
        class="_social-icons_acfd2c"
      >
        <div
          class="_social-icon_acfd2c"
        >
          <svg
            class="tabler-icon tabler-icon-brand-facebook "
            fill="none"
            height="18"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 10v4h3v7h4v-7h3l1 -4h-4v-2a1 1 0 0 1 1 -1h3v-4h-3a5 5 0 0 0 -5 5v2h-3"
            />
          </svg>
        </div>
        <div
          class="_social-icon_acfd2c"
        >
          <svg
            class="tabler-icon tabler-icon-brand-tiktok "
            fill="none"
            height="18"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 7.917v4.034a9.948 9.948 0 0 1 -5 -1.951v4.5a6.5 6.5 0 1 1 -8 -6.326v4.326a2.5 2.5 0 1 0 4 2v-11.5h4.083a6.005 6.005 0 0 0 4.917 4.917z"
            />
          </svg>
        </div>
        <div
          class="_social-icon_acfd2c"
        >
          <svg
            class="tabler-icon tabler-icon-brand-x "
            fill="none"
            height="18"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 4l11.733 16h4.267l-11.733 -16z"
            />
            <path
              d="M4 20l6.768 -6.768m2.46 -2.46l6.772 -6.772"
            />
          </svg>
        </div>
        <div
          class="_social-icon_acfd2c"
        >
          <svg
            class="tabler-icon tabler-icon-brand-instagram "
            fill="none"
            height="18"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 8a4 4 0 0 1 4 -4h8a4 4 0 0 1 4 4v8a4 4 0 0 1 -4 4h-8a4 4 0 0 1 -4 -4z"
            />
            <path
              d="M9 12a3 3 0 1 0 6 0a3 3 0 0 0 -6 0"
            />
            <path
              d="M16.5 7.5v.01"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="_details-container_acfd2c m_6d731127 mantine-Stack-root"
      style="--stack-gap: calc(1rem * var(--mantine-scale)); --stack-align: stretch; --stack-justify: flex-start;"
    >
      <h3
        class="_section-title_acfd2c"
      >
        Navigation
      </h3>
      <div
        class="_line_acfd2c"
      />
      <div
        class="_nav-links_acfd2c"
      >
        <p>
          About Us
        </p>
        <p>
          Hotels
        </p>
        <p>
          FAQs
        </p>
        <p>
          My Bookings
        </p>
      </div>
    </div>
    <div
      class="_details-container_acfd2c m_6d731127 mantine-Stack-root"
      style="--stack-gap: calc(1rem * var(--mantine-scale)); --stack-align: stretch; --stack-justify: flex-start;"
    >
      <h3
        class="_section-title_acfd2c"
      >
        Legal
      </h3>
      <div
        class="_line_acfd2c"
      />
      <div
        class="_legal-links_acfd2c"
      >
        <p>
          Terms of Service
        </p>
        <p>
          Privacy Policy
        </p>
        <p>
          Cookie Policy
        </p>
      </div>
    </div>
  </div>
</div>
`;
